tests/unit/config/test_configfiles.py

```diff
@@ -879,6 +879,40 @@ def test_content_media_capture(self, yaml, autoconfig):
                         'content.media.video_capture']:
             assert data[setting]['global'] == val
 
+    @pytest.mark.parametrize('chars, uppercase', [
+        ('aoeuidnths', True),
+        ('abcdefghijklmnopqrstuvwxyz', True),
+        (None, True),
+        ('aoeuidnths', False),
+        ('abcdefghijklmnopqrstuvwxyz', False),
+        (None, False),
+        ('aoeuidnths', None),
+        ('abcdefghijklmnopqrstuvwxyz', None),
+        (None, None),
+    ])
+    def test_hints_uppercase(self, yaml, autoconfig, chars, uppercase):
```

Comment:
I'm not super happy with this test. I dislike the use `None` values, but I couldn't figure out how to do this in a nicer way. The problem is the interdependence of `hints.chars` and `hints.uppercase`. We have to test for the possibility that one or both of them is unset. Is there a better way to represent this than the use of `None` values?